xmodule/modulestore/split_mongo/split.py

```diff
@@ -2438,23 +2437,14 @@ def _copy_from_template(
 
         for usage_key in source_keys:
             src_course_key = usage_key.course_key
-            hashable_source_id = src_course_key.for_version(None)
             block_key = BlockKey(usage_key.block_type, usage_key.block_id)
             source_structure = source_structures[src_course_key]
 
             if block_key not in source_structure['blocks']:
                 raise ItemNotFoundError(usage_key)
             source_block_info = source_structure['blocks'][block_key]
 
-            # Compute a new block ID. This new block ID must be consistent when this
-            # method is called with the same (source_key, dest_structure) pair
-            unique_data = "{}:{}:{}".format(
-                str(hashable_source_id).encode("utf-8"),
-                block_key.id,
-                new_parent_block_key.id,
-            )
-            new_block_id = hashlib.sha1(unique_data.encode('utf-8')).hexdigest()[:20]
-            new_block_key = BlockKey(block_key.type, new_block_id)
+            new_block_key = derived_key(src_course_key, block_key, new_parent_block_key)
```

Comment:
Moving `derived_key` to a separate function does seem a bit cleaner, but what's the motivation here? It's only used in this one place still as far as I can see.